.github/workflows/coverage.yml

```diff
@@ -0,0 +1,46 @@
+#
+# SPDX-License-Identifier: Apache-2.0
+# SPDX-FileCopyrightText: Huawei Inc.
+#
+name: xpanse-test-coverage
+
+on:
+  push:
+    branches: [main]
+  pull_request:
+    branches: [main]
+
+jobs:
```

Comment:
we must also ensure that the job runs only on actual repo and not on forked repos.
Please add a check like this -> https://github.com/eclipse-xpanse/xpanse/blob/main/.github/workflows/release.yml#L26